javascript/javascript2/week3/Test.js

```diff
@@ -0,0 +1,217 @@
+//Call the function you have created with some different arguments.
+window.addEventListener("load", logOutString);
+function logOutString(delay, stringToLog) {
+  const num = delay * 1000;
+  setTimeout(function () {
+    const paragraphSpace = document.getElementById("class1");
+    paragraphSpace.innerHTML = `${delay}, ${stringToLog}`;
+    //window.alert(`${delay}, ${stringToLog}`);
+  }, num);
+}
+logOutString(5, "The String Logged after 5 seconds");
+logOutString(3, "The String Logged after 3 seconds");
+
+//Create a button in html. When clicking this button, use the function you created in the previous task to log out the text: Called after 5 seconds 5 seconds after the button is clicked.
+const newButton = document.getElementById("first"); // add function above
+newButton.addEventListener("click", eventOut1);
+
+function eventOut1() {
+  setTimeout(function () {
+    const paragraphSpace = document.getElementById("class2");
+    paragraphSpace.innerHTML = "Called after 2.5 seconds";
+  }, 2500);
+}
+// Create two functions and assign them to two different variables. One function logs out Earth, the other function logs out Saturn.
+
+const earthLog = () => {
+  console.log("Earth");
+};
+const saturnLog = () => {
+  console.log("Saturn");
+};
+
+function planetLog(planetLogFunc) {
+  planetLogFunc();
+}
+planetLog(earthLog);
+planetLog(saturnLog);
+
+//Create a button with the text called "Log location". When this button is clicked the location (latitude, longitude) of the user should be logged out
+
+const locationLogButton = document.getElementById("loca");
+const displayLoca = document.getElementById("displayLoca");
+locationLogButton.addEventListener("click", displayLatLon);
+
+function displayLatLon() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(showPosition);
+  } else {
+    displayLoca.innerHTML = "Geolocation is not supported by this browser.";
+  }
+}
+
+function showPosition(position) {
+  displayLoca.innerHTML =
+    "Latitude: " +
+    position.coords.latitude +
+    "<br>Longitude: " +
+    position.coords.longitude;
+}
+
+// Create a function called runAfterDelay. It has two parameters: delay and callback. When called the function should wait delay seconds and then call the provided callback function.
+//Try and call this function with different delays and different callback functions
+
+window.addEventListener("load", runAfterDelay);
+function runAfterDelay(delays, seeString) {
+  let delaysIn = delays * 1000;
+  setTimeout(seeString, delaysIn);
+}
+// runAfterDelay(4, function seeString() {
+//   console.log("Should be logged after 4 seconds");
+// });
+
+//Check if we have double clicked on the page. A double click is defined by two clicks within 0.5 seconds.
+//If a double click has been detected, log out the text: "double click!"
+
+const space = document.getElementById("headsup");
+space.addEventListener("click", doubleClick, true);
+let clickedOnce = new Date().getMilliseconds();
```

Comment:
This is out side of your function and it will get called when your .js file load. so lets say if you load a file 15 min before calling the below function doubleClick() how you will check the first click was before 0.5 sec ?

Reply:
can you please ellaborate because I havent understood your question?